examples/kokkos_assembly/kokkos_assembly.cpp

```diff
@@ -0,0 +1,241 @@
+    md.sum_duplicates();
+
+    // Build Csr matrix.
+    matrix->read(std::move(md));
+}
+
+
+// Generates the RHS vector given `f` and the boundary conditions.
+template <typename Closure, typename ValueType>
+void generate_rhs(Closure&& f, ValueType u0, ValueType u1,
+                  gko::matrix::Dense<ValueType>* rhs)
+{
+    const auto discretization_points = rhs->get_size()[0];
+    auto values = rhs->get_values();
+    Kokkos::View<ValueType*> values_view(values, discretization_points);
+    Kokkos::parallel_for(
+        "generate_rhs", discretization_points, KOKKOS_LAMBDA(int i) {
+            const ValueType h = 1.0 / (discretization_points + 1);
+            const ValueType xi = ValueType(i + 1) * h;
+            values_view[i] = -f(xi) * h * h;
+            if (i == 0) {
+                values_view[i] += u0;
+            }
+            if (i == discretization_points - 1) {
+                values_view[i] += u1;
+            }
+        });
+}
+
+
+// Prints the solution `u`.
+template <typename ValueType>
+void print_solution(ValueType u0, ValueType u1,
+                    const gko::matrix::Dense<ValueType>* u)
+{
+    auto host_u = gko::make_temporary_clone(u->get_executor()->get_master(), u);
+    std::cout << u0 << '\n';
+    for (int i = 0; i < u->get_size()[0]; ++i) {
+        std::cout << host_u->get_const_values()[i] << '\n';
+    }
+    std::cout << u1 << std::endl;
+}
+
+
+// Computes the 1-norm of the error given the computed `u` and the correct
+// solution function `correct_u`.
+template <typename Closure, typename ValueType>
+double calculate_error(int discretization_points,
+                       const gko::matrix::Dense<ValueType>* u,
+                       Closure&& correct_u)
+{
+    Kokkos::View<const ValueType*> v_u(u->get_const_values(),
+                                       discretization_points);
+    auto error = 0.0;
+    Kokkos::parallel_reduce(
+        "calculate_error", discretization_points,
+        KOKKOS_LAMBDA(int i, double& lsum) {
+            const auto h = 1.0 / (discretization_points + 1);
+            const auto xi = (i + 1) * h;
+            lsum += Kokkos::Experimental::abs(
+                (v_u(i) - correct_u(xi)) /
+                Kokkos::Experimental::abs(correct_u(xi)));
+        },
+        error);
+    return error;
+}
+
+
+int main(int argc, char* argv[])
+{
+    Kokkos::ScopeGuard kokkos(argc, argv);
+
+    // Some shortcuts
+    using ValueType = double;
+    using RealValueType = gko::remove_complex<ValueType>;
+    using IndexType = int;
+
+    using vec = gko::matrix::Dense<ValueType>;
+    using mtx = gko::matrix::Csr<ValueType, IndexType>;
+    using cg = gko::solver::Cg<ValueType>;
+    using bj = gko::preconditioner::Jacobi<ValueType>;
+
+    // Figure out where to run the code
+    if (argc == 2 && (std::string(argv[1]) == "--help")) {
+        std::cerr << "Usage: " << argv[0]
+                  << " [discretization_points] [kokkos_assembly-options]"
```

Comment:
What are the kokkos assembly options ?

Reply:
The kokkos CLI options. I will add a link to that.

Reply:
Does that work even if in this case you have `discretization_points` as the first argument ? I assume the kokkos scoped guard uses argc and argv to parse that. If they dont care about the order, then I guess it should be fine.